packages/wallets/coin98/src/adapter.ts

```diff
@@ -158,7 +158,37 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
         const signedTransactions: Transaction[] = [];
         for (const transaction of transactions) {
             signedTransactions.push(await this.signTransaction(transaction));
+            await this.sleep()
         }
         return signedTransactions;
     }
+
+
+    async signMessage(message: Uint8Array): Promise<Uint8Array> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                const decodedMessage = new TextDecoder().decode(message)
+                
+                const response = await wallet.request({ method: 'sol_sign', params: [decodedMessage] });
+
+                const signature = new TextEncoder().encode(response.sig);
+
+                return signature;
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    //Ultilities
+    sleep(ms = 500): Promise<boolean>{
+        return new Promise(resolve => setTimeout(resolve, ms))
+    }
```

Comment:
Please remove, timeouts are not a reliable, deterministic source of async evaluation. I'm also not sure what the delay is for.